rules/python/shared/lang/dynamic_input.yml

```diff
@@ -0,0 +1,33 @@
+type: shared
+languages:
+  - python
+patterns:
+  - sys.argv[$<_>]
+  - pattern: $<PARSER>.parse_args($<...>)
+    filters:
+      - variable: PARSER
+        detection: python_shared_lang_dynamic_input_parser
+        scope: result
+auxiliary:
+  - id: python_shared_lang_dynamic_input_parser
+    patterns:
+      - pattern: $<ARG_PARSER>()
+        filters:
+          - variable: ARG_PARSER
+            detection: python_shared_lang_dynamic_input_argument_parser
+  - id: python_shared_lang_dynamic_input_argument_parser
+    patterns:
+      - ArgumentParser # fallback
+      - from argparse import ArgumentParser as $<!>$<_>
```

Comment:
This isn't matching...

It compiles to the following, which matches as expected on Tree Sitter 🤔 

```
([(import_from_statement  . [(dotted_name . [ (identifier )] @param1 .)]  . [(aliased_import . [(dotted_name . [ (identifier )] @param2 .)]  . alias: (_) @match .)] .)] @root)
```

Reply:
^ https://github.com/Bearer/bearer/pull/1592